### ISSUE TITLE
[Feature request] Different exit code depending on only warnings or error

### ISSUE THREAD

**Issue by VRehnberg** [linkchecker/linkchecker#844]
## Summary
Currently exit code is 1 if warnings or errors are detected. 

I'm using `linkchecker` as part of a CI pipeline for checking links after building a public website and would like to handle warnings and errors separately.

## Steps to reproduce

Consider the following three examples
 1. `linkchecker https://github.com/linkchecker/linkchecker`
 2. `linkchecker http://github.com/linkchecker/linkchecker`
 3. `linkchecker https://github.com/linkchecker/linkchecker-not-exist`
and check the exit codes `echo $?`.

## Actual result
Currently you get
1. `0`
2. `1`
3. `1`

## Expected result
While I would like it to be:
```
1. `0`
2. `1`
3. Something >1
```

## Environment
Don't think this is relevant.
 * Operating system: Linux c3se-TP6 6.8.0-45-generic #45~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Wed Sep 11 15:25:05 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
 * Linkchecker version: LinkChecker 10.5.0 released 2024-09-03
Copyright (C) 2000-2016 Bastian Kleineidam, 2010-2024 LinkChecker Authors
 * Python version: Python 3.12.5
 * Install method: `conda install pip; pip install linkchecker`

## Logs
```
(deadlinks) vikren@c3se-TP6:~$ linkchecker https://github.com/linkchecker/linkchecker -r 0; echo $?
INFO linkcheck.cmdline 2024-09-30 15:16:33,595 MainThread Checking intern URLs only; use --check-extern to check extern URLs.
LinkChecker 10.5.0
Copyright (C) 2000-2016 Bastian Kleineidam, 2010-2024 LinkChecker Authors
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it under
certain conditions. Look at the file `COPYING' within this distribution.
Read the documentation at https://linkchecker.github.io/linkchecker/
Write comments and bugs to https://github.com/linkchecker/linkchecker/issues

Start checking at 2024-09-30 15:16:33+002

Statistics:
Downloaded: 0B.
Content types: 0 image, 1 text, 0 video, 0 audio, 0 application, 0 mail and 0 other.
URL lengths: min=42, max=42, avg=42.

That's it. 1 link in 1 URL checked. 0 warnings found. 0 errors found.
Stopped checking at 2024-09-30 15:16:35+002 (1 seconds)
0
(deadlinks) vikren@c3se-TP6:~$ linkchecker http://github.com/linkchecker/linkchecker -r 0; echo $?
INFO linkcheck.cmdline 2024-09-30 15:16:43,204 MainThread Checking intern URLs only; use --check-extern to check extern URLs.
LinkChecker 10.5.0
Copyright (C) 2000-2016 Bastian Kleineidam, 2010-2024 LinkChecker Authors
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it under
certain conditions. Look at the file `COPYING' within this distribution.
Read the documentation at https://linkchecker.github.io/linkchecker/
Write comments and bugs to https://github.com/linkchecker/linkchecker/issues

Start checking at 2024-09-30 15:16:43+002

URL        `http://github.com/linkchecker/linkchecker'
Real URL   https://github.com/linkchecker/linkchecker
Check time 0.588 seconds
Warning    [http-redirected] Redirected to
           `https://github.com/linkchecker/linkchecker' status:
           301 Moved Permanently.
Result     Valid: 200 OK

Statistics:
Downloaded: 0B.
Content types: 0 image, 1 text, 0 video, 0 audio, 0 application, 0 mail and 0 other.
URL lengths: min=42, max=42, avg=42.

That's it. 1 link in 2 URLs checked. 1 warning found. 0 errors found.
Stopped checking at 2024-09-30 15:16:44+002 (1 seconds)
1
(deadlinks) vikren@c3se-TP6:~$ linkchecker https://github.com/linkchecker/linkchecker-not-exist -r 0; echo $?
INFO linkcheck.cmdline 2024-09-30 15:16:57,111 MainThread Checking intern URLs only; use --check-extern to check extern URLs.
LinkChecker 10.5.0
Copyright (C) 2000-2016 Bastian Kleineidam, 2010-2024 LinkChecker Authors
LinkChecker comes with ABSOLUTELY NO WARRANTY!
This is free software, and you are welcome to redistribute it under
certain conditions. Look at the file `COPYING' within this distribution.
Read the documentation at https://linkchecker.github.io/linkchecker/
Write comments and bugs to https://github.com/linkchecker/linkchecker/issues

Start checking at 2024-09-30 15:16:57+002

URL        `https://github.com/linkchecker/linkchecker-not-exist'
Real URL   https://github.com/linkchecker/linkchecker-not-exist
Check time 0.776 seconds
Result     Error: 404 Not Found

Statistics:
Downloaded: 0B.
Content types: 0 image, 1 text, 0 video, 0 audio, 0 application, 0 mail and 0 other.
URL lengths: min=52, max=52, avg=52.

That's it. 1 link in 1 URL checked. 0 warnings found. 1 error found.
Stopped checking at 2024-09-30 15:16:58+002 (1 seconds)
1
```

## Other notes

All I care about is that warnings and errors can be differentiated based on the error code. There are some light conventions on exit codes see for example https://tldp.org/LDP/abs/html/exitcodes.html, but they seem quite loose.
